packages/imperative/src/config/src/ProfileInfo.ts

```diff
@@ -1225,6 +1234,301 @@ export class ProfileInfo {
         LoggerUtils.setProfileSchemas(this.mProfileSchemaCache);
     }
 
+    /**
+     * Reads the `extenders.json` file from the CLI home directory.
+     * Called once in `readProfilesFromDisk` and cached to minimize I/O operations.
+     */
+    private readExtendersJsonFromDisk(): void {
```

Comment:
Unfortunately, since I made it static to access during plugin installation, I cannot access the `ensureReadFromDisk` function. But, the function will ensure that the `extenders.json` file is created if it does not exist at the global level, and it will fall back to the global-level file if it doesn't exist at the project level.